internal/config/config.go

```diff
@@ -471,9 +499,81 @@ func addressPoolFromCR(p metallbv1beta1.IPAddressPool) (*Pool, error) {
 		ret.cidrsPerAddresses[cidr] = nets
 	}
 
+	serviceAllocations, err := addressPoolServiceAllocationsFromCR(p, namespaces)
+	if err != nil {
+		return nil, err
+	}
+	ret.ServiceAllocations = serviceAllocations
+
 	return ret, nil
 }
 
+func addressPoolServiceAllocationsFromCR(p metallbv1beta1.IPAddressPool, namespaces []corev1.Namespace) (*ServiceAllocation, error) {
+	if p.Spec.AllocateTo == nil ||
+		(len(p.Spec.AllocateTo.Namespaces) > 0 && len(p.Spec.AllocateTo.NamespaceSelectors) > 0 &&
+			len(p.Spec.AllocateTo.ServiceSelectors) > 0) {
+		return nil, nil
+	}
+	serviceAllocations := &ServiceAllocation{}
+	serviceAllocations.Priority = p.Spec.AllocateTo.Priority
+	if len(p.Spec.AllocateTo.Namespaces) > 0 || len(p.Spec.AllocateTo.NamespaceSelectors) > 0 {
+		serviceAllocations.Namespaces = sets.NewString()
+	}
+	for _, namespace := range p.Spec.AllocateTo.Namespaces {
```

Comment:
Thinking out loudly as it may impact the implementation, but would it be more efficient if we propagate the namespace selector and pass the whole namespace object (or just its labels) to the allocator?
Here we iterate on all the namespaces every time there's a config change. Maybe it's not that much.